donate/payments/forms.py

```diff
@@ -157,11 +170,21 @@ class CurrencyForm(forms.Form):
 
 class UpsellForm(MinimumCurrencyAmountMixin, forms.Form):
     currency = forms.ChoiceField(choices=constants.CURRENCY_CHOICES, widget=forms.HiddenInput, disabled=True)
+    frequency = forms.ChoiceField(
+        choices=constants.FREQUENCY_CHOICES,
+        widget=forms.HiddenInput,
+        initial=constants.FREQUENCY_MONTHLY,
+        required=False,
+    )
     amount = forms.DecimalField(
         label=_('Amount'), min_value=1, max_value=MAX_AMOUNT_VALUE, decimal_places=2,
         widget=forms.NumberInput(attrs={'step': 'any'})
     )
 
+    def clean_frequency(self):
```

Comment:
**Question** Hey Tibor, I see that this `clean_frequency` is defined, but do not see it being used anywhere else. Just wanted to double check that this was intended?

Reply:
Yea, this is a bit of implicit functionality that comes in by extending the Django `forms.Form`. The `clean_frequency` method is triggered under the hood when `is_valid` is called on the form (so are all other methods that follow the pattern `clean_<fieldname>`). 

There is a whole page in the Django docs dedicated to the form and field validation. I guess that's a pointer that this raises questions quite frequently 😅 
https://docs.djangoproject.com/en/4.1/ref/forms/validation/#form-and-field-validation